src/main/java/com/ericsson/ei/subscription/SubscriptionValidator.java

```diff
@@ -180,7 +185,12 @@ private static void RestPostMediaType(String restPostMediaType) throws Subscript
      * @param email
      */
     public static void validateEmail(String email) throws SubscriptionValidationException {
-        final Pattern VALID_EMAIL_ADDRESS_REGEX = Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$",
+        // When this regExp need to be changed then remember to change the one in the
+        // back-end (validEmailRegExpression in subscription.js), which do the same
+        // email validation check. The two
+        // regEx always need to be the same for ensuring the same check.
+        String validEmailRegExpression = "^(([^<>()\\[\\]\\\\.,;:\\s@\"]+(\\.[^<>()\\[\\]\\\\.,;:\\s@\"]+)*)|(\".+\"))@((\\[[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}])|(([a-zA-Z\\-0-9]+\\.)+[a-zA-Z]{2,}))$";
```

Comment:
This is a very very long regex! Have you unit tests for all cases this regex should match and fail on?
Is this supposed to regex on both list of mails (comma separated) and a single mail.
I think the code becomes a lot more readable if you have regexp for a single mail and them split the mail-list on comma and validate one mail at time.
Now not human understandable regexp for a normal developer.
What is valid characters in mail: '.', '-', '@' , '[a-zA-Z0-9]',,, am I missing something?

Reply:
Well, unit tests are already there, I just took the regex(the longer one) from frontend and put in the backend. 
In fact, it is for single email validation. But, I agree it is difficult to comprehend. We will discuss it before taking next step.

Reply:
Of course this is a long string, this is a string created by communities for valid email addresses. We shouldn't make them ourselves.